clients/fides-js/src/fides-tcf.ts

```diff
@@ -185,14 +185,15 @@ const init = async (config: FidesConfig) => {
     updateExperience,
     overrides,
   });
-  Object.assign(_Fides, updatedFides);
+  Object.assign(this, updatedFides);
```

Comment:
aside from the `this` conversation...it also seems odd to me that we're mixing both spread operators and Object.assign in the same codebase. Couldn't we just use spread like config does, or am I missing something?
```suggestion
  this = {...this, ...updatedFides};
```

Maybe `Object.assign` was originally used to avoid the need for `// eslint-disable-next-line no-param-reassign`?

Consistency is my main concern.

Reply:
Ah, spreading wouldn't work with the current code because you cannot replace `this` with a new value. I used spread with the `config` to avoid modifying the argument passed to the function because it's what gets stored in `prevConfig`. Otherwise, using `reinit()` would call `init` with slightly different arguments than it was initially called with, which, while it worked fine locally, is not something I'd want other developers to discover and have to keep in mind when working on this.

Reply:
Thanks for that explanation. So if I understand correctly, spreading would have been ok with the original code, but the `this` usage makes it inappropriate.